src/proxy/hdrs/XPACK.cc

```diff
@@ -559,7 +578,47 @@ XpackDynamicTableStorage::write(const char *name, uint32_t name_len, const char
 }
 
 void
-XpackDynamicTableStorage::erase(uint32_t name_len, uint32_t value_len)
+XpackDynamicTableStorage::erase(uint32_t to_erase)
+{
+  // Ensure that _tail is not overcoming _head.
+  ink_release_assert(to_erase <= this->size());
```

Comment:
For the same reason above, I think we should avoid a release assert. See https://github.com/apache/trafficserver/issues/10082